src/plugins/data_source_management/public/plugin.ts

```diff
@@ -303,10 +319,85 @@ export class DataSourceManagementPlugin
     };
     setRenderAssociatedObjectsDetailsFlyout(renderAssociatedObjectsDetailsFlyout);
 
+    const getDirectQuerySyncFromUrl = (): boolean | null => {
+      const parsedHash = parseUrlHash(window.location.href);
+      if (parsedHash && parsedHash.query) {
+        const directQuerySync = parsedHash.query.directQuerySync;
+        if (directQuerySync === 'true') return true;
+        if (directQuerySync === 'false') return false;
+      }
+      return null;
+    };
+
+    const shouldMountBanner = (): boolean => {
+      const urlFlag = getDirectQuerySyncFromUrl();
+      if (urlFlag !== null) {
+        return urlFlag;
```

Comment:
is this what we want.

should it not be like this config is to ensure that if we need to shut it off then no matter what the url passes in it will be ignored?

like should it not be something like

```typescript
return this.config.dashboardDirectQuerySyncEnabled && getDirectQuerySyncFromUrl()
```

(I dont think that is accurate return but somethjing similar to that)

Reply:
Considering the safe deployment, we would like to have the feature flag in config `off	` by default. And for CW console to use it, it will just pass that url flag to `true` to enabled it in their iframe embeddable.